### PR TITLE
Standardise MAP Timestamps

### LINKED ISSUE #5150: ARES timestamps are sometimes in seconds; sometimes in milliseconds
**Describe the bug**
ARES sometimes returns timestamps in seconds - sometimes in milliseconds

Example 1: https://www.test.bbc.com/pidgin/23248703
Example 2: https://www.stage.bbc.com/pidgin/media-43441368?renderer_env=live

**Expected behavior**
We show the correct time, whether the timestamp is in seconds or milliseconds

It seems safe to assume that, if a timestamp resolves to 1970, it needs converted to milliseconds

**Additioanal context (internal links)**
https://bbcnews.slack.com/archives/CJT0284G2/p1579016012090800
https://bbcnews.slack.com/archives/C70C1RPGE/p1579019279014600

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves https://github.com/bbc/simorgh/issues/5150

**Overall change:** 
Update the logic we use to convert timestamps within metadata.

This was not always working - the ares version does not seem to be a reliable way to determine whether the `lastUpdated` timestamp is in seconds or milliseconds

Also applied this logic to the related content timestamps

**Code changes:**
- _Assume timestamps are in seconds if they're below a certain value_

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/containers/CpsRelatedContent/index.test.jsx [hunk 1]
```diff
@@ -63,4 +66,35 @@ describe('CpsRelatedContent', () => {
...
+    );
+
+    renderRelatedContent({
+      content: transformedPromos,
+    });
+
+    const renderedTime = document
+      .querySelector(`aside time`)
+      .getAttribute('datetime');
+    expect(renderedTime).toBe('2009-02-14');
```
Comment (rebeccamcginn) [inline src/app/containers/CpsRelatedContent/index.test.jsx]: Does 1234567890/1234567890000 not convert to Friday, 13 February 2009 23:31:30?

Comment (ryanmccombe) [reply src/app/containers/CpsRelatedContent/index.test.jsx]: Umm, good point - the test passes, so maybe a timezone or daylight savings thing?

Need to investigate a bit more - don't want a test that fails depending on what date it is run 😬

Comment (rebeccamcginn) [reply src/app/containers/CpsRelatedContent/index.test.jsx]: A job for tomorrow me thinks 🙂

**Review by jroebu14 — COMMENTED** [review]
In our tests, can we use functions like [`getByText`](https://github.com/testing-library/react-testing-library/blob/fd2df8d18652786a95bce34741180137f9d2cef2/README.md#getbytexttext-textmatch-htmlelement) or any other of the methods returned from the `render` function where possible?

I think we should stick to the [guiding principles](https://github.com/testing-library/react-testing-library/blob/fd2df8d18652786a95bce34741180137f9d2cef2/README.md#guiding-principles) of the test renderer we have adopted.

> In the spirit of the guiding principles, it is recommended to use `getByTestId` only after `getByLabel`, `getByPlaceholderText` or `getByText` don't work for your use case. Using `data-testid` attributes do not resemble how your software is used and should be avoided if possible. That said, they are way better than querying based on DOM structure. Learn more about data-testids from the blog post ["Making your UI tests resilient to change"](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)

File: src/app/containers/CpsRelatedContent/index.test.jsx (hunk 1 above)
Comment (rhenshaw56) [reply src/app/containers/CpsRelatedContent/index.test.jsx]: not sure if this is related but we had a similar issue with  the article timestamp tests failing on ci at odd hours in the morning, the [fix](https://github.com/bbc/simorgh/pull/4703/files) was to stub `Date.now` to force it to use the exact time we wanted

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/timestampToMilliseconds/index.js [hunk 2]
```diff
@@ -1,33 +1,54 @@
...
-  const semverComparison = semverCompare(aresVersion, '1.1.0');
-
-  // if ares version is 1.1.0 or above, do nothing
-  if (semverComparison !== -1) return jsonRaw;
+const standardiseMetadataTimestamps = jsonRaw => {
+  const firstPublished = path(['metadata', 'firstPublished'], jsonRaw);
+  const lastPublished = path(['metadata', 'lastPublished'], jsonRaw);
+  const lastUpdated = path(['metadata', 'lastUpdated'], jsonRaw);
 
   const json = deepClone(jsonRaw);
```
Comment (jroebu14) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/timestampToMilliseconds/index.js]: a suggestion without the deep clone with a couple of useful functions that could be used further down

```js
const isInSeconds = timestamp => timestamp < MINIMUM_TIMESTAMP_VALUE;

const standardiseTimestamp = timestamp =>
  isInSeconds(timestamp) ? convertToMilliseconds(timestamp) : timestamp;

const standardiseMetadataTimestamps = jsonRaw => {
  const firstPublishedInMs = standardiseTimestamp(jsonRaw.firstPublished);
  const lastPublishedInMs = standardiseTimestamp(jsonRaw.lastPublished);
  const lastUpdatedInMs = standardiseTimestamp(jsonRaw.lastUpdated);

  return mergeDeepLeft(
    {
      metadata: {
        firstPublished: firstPublishedInMs,
        lastPublished: lastPublishedInMs,
        lastUpdated: lastUpdatedInMs,
      },
    },
    jsonRaw,
  );
};
```

sorry I'm at a loose end this morning!

File: src/app/containers/CpsRelatedContent/index.test.jsx (hunk 1 above)
Comment (ryanmccombe) [reply src/app/containers/CpsRelatedContent/index.test.jsx]: 👍 Updated test to just check the month and year - not the date

**Review by jroebu14 — APPROVED** [review]
Looks good to me. Thanks for making the changes

File: src/app/containers/CpsRelatedContent/index.test.jsx [hunk 3]
```diff
@@ -88,13 +88,10 @@ describe('CpsRelatedContent', () => {
...
-    renderRelatedContent({
+    const { getByText } = renderRelatedContent({
       content: transformedPromos,
     });
 
-    const renderedTime = document
-      .querySelector(`aside time`)
-      .getAttribute('datetime');
-    expect(renderedTime).toBe('2009-02-14');
+    expect(getByText('February 2009', { exact: false })).not.toBeNull();
```
Comment (rebeccamcginn) [inline src/app/containers/CpsRelatedContent/index.test.jsx:95]: expect(getByText('February 2009')).toBeInTheDocument();

Would this work? I'm expecting @jroebu14 to suggest it :)

Comment (ryanmccombe) [reply src/app/containers/CpsRelatedContent/index.test.jsx:95]: I had that originally, but `toBeInTheDocument` is seemingly not part of react testing library - it comes from another package: https://stackoverflow.com/questions/56547215/react-testing-library-why-is-tobeinthedocument-not-a-function

Comment (rebeccamcginn) [reply src/app/containers/CpsRelatedContent/index.test.jsx:95]: Interesting. Thanks for explaining :)

**Review by rebeccamcginn — APPROVED** [review]

**Pull request merged**